src/main/scala/com/rbmhtechnology/calliope/javadsl/EventStore.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2015 - 2017 Red Bull Media House GmbH <http://www.redbullmediahouse.com> - all rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.rbmhtechnology.calliope.javadsl
+
+import java.lang.{Iterable => JIterable}
+
+import com.rbmhtechnology.calliope.{StoredEvent, scaladsl}
+
+import scala.collection.immutable
+
+trait EventStore[A] {
```

Comment:
Yes, but it is the only indicator for the target-type of the `TransactionalEventProducer`. We could also remove the type-parameter here and require it on creation of the producer.
WDYT?